architecture/configuration.md

```diff
@@ -0,0 +1,58 @@
+# Spike Config
+
+Most of the inner workings of `spike` will never be open to configuration, we
+allow the users a limited interface for changing some of the default behavior of
+`spike` for a smoother experience.
+
+This configuration will be handled by a configuration object, that will exist
+under a `"spike"` key in the users `package.json` or a `spike.config.js` file in
+the source directory.
+
+
+## Scope
+
+The configuration file will allow you to change the following options:
+
+- Changing the build target directory. Optional, default to `out/`.
+- Changing the source directory. Optional, default to `cwd`.
+- Enabling typescript support. Optional, default to `false`.
```

Comment:
In addition, this needs to clearly be marked as an "experimental" feature

Reply:
we can probably follow nextjs convention here as well and have an `experimental` key under which you can enable experimental features. or maybe `dangerous`. i want to make it clear that by opting in to this, you are opting out of the "no breaking changes ever" guarantee. or we could just cut it from the standard config and require ejection, which is the standard mechanism by which users out out of the no breaking changes ever guarantee

Reply:
I _love_ this idea.